src/proto/streams/send.rs

```diff
@@ -339,7 +343,7 @@ impl Send {
         if available as usize <= buffered {
             0
         } else {
-            available - buffered as WindowSize
+            available.min(self.max_buffer_size as WindowSize) - buffered as WindowSize
```

Comment:
I think this can underflow, right?

Reply:
Never mind, clearly I had not woken up correctly yet.

Reply:
No, you were right. I add a test, and it did panic with an overflow. I've fixed that.